sdk/core/azure-core/azure/core/pipeline/policies/_utils.py

```diff
@@ -66,7 +66,8 @@ def parse_retry_after(retry_after: str) -> float:
     """
     delay: float  # Using the Mypy recommendation to use float for "int or float"
     try:
-        delay = float(retry_after)
+        retry_after_clean = retry_after.replace(",", "")
```

Comment:
One thing about this issue that I'm wondering --> is the comma used for formatting or a [decimal separator](https://en.wikipedia.org/wiki/Decimal_separator).

81,000 becomes 81000 and then /1000 gives 81 secs for the retry time. Luckily this scenario works for comma used either as formatting or decimal point. The other case of 1,2 (1.2) but return as 12 would be totally different values.

Reply:
My understanding is that we only need to care about the retry-after header returning from Azure services.

And I assume our architects will stop the service teams from returning those formats? :)

Reply:
@marseel Thanks for the information.

Do you happen to know the header?

e.g. is it "retry-after", "retry-after-ms","x-ms-retry-after-ms" or something else?

Reply:
Unfortunately, I didn't see it anywhere in our or Azure logs.

I guess we could figure it out by checking azure-cli version / SDK version and taking a look at traceback which one it is.

Reply:
If you can add code like

import logging
logging.basicConfig(filename='your path', level=logging.DEBUG)

we may be able to collect the headers.

Or can you share the repro steps?